src/goInstallTools.ts

```diff
@@ -11,38 +11,67 @@ import path = require('path');
 import os = require('os');
 import cp = require('child_process');
 import { showGoStatus, hideGoStatus } from './goStatus';
-import { getBinPath } from './goPath';
+import { getBinPath, getGoRuntimePath } from './goPath';
 import { outputChannel } from './goStatus';
 
-let tools: { [key: string]: string } = {
-	'gocode': 'github.com/nsf/gocode',
-	'goreturns': 'sourcegraph.com/sqs/goreturns',
-	'gopkgs': 'github.com/tpng/gopkgs',
-	'godef': 'github.com/rogpeppe/godef',
-	'golint': 'github.com/golang/lint/golint',
-	'go-outline': 'github.com/lukehoban/go-outline',
-	'go-symbols': 'github.com/newhook/go-symbols',
-	'guru': 'golang.org/x/tools/cmd/guru',
-	'gorename': 'golang.org/x/tools/cmd/gorename',
-	'goimports': 'golang.org/x/tools/cmd/goimports'
-};
+interface SemVersion {
+	major: number;
+	minor: number;
+}
+
+let goVersion: SemVersion = null;
+
+function getTools(): { [key: string]: string }  {
+	let goConfig = vscode.workspace.getConfiguration('go');
+	let tools: { [key: string]: string } = {
+		'gocode': 'github.com/nsf/gocode',
+		'gopkgs': 'github.com/tpng/gopkgs',
+		'godef': 'github.com/rogpeppe/godef',
+		'go-outline': 'github.com/lukehoban/go-outline',
+		'go-symbols': 'github.com/newhook/go-symbols',
+		'guru': 'golang.org/x/tools/cmd/guru',
+		'gorename': 'golang.org/x/tools/cmd/gorename'
+	};
+
+	// Install the formattool that was chosen by the user
+	if (goConfig['formatTool'] === 'goimports') {
+		tools['goimports'] = 'golang.org/x/tools/cmd/goimports';
+	} else if (goConfig['formatTool'] === 'goreturns') {
+		tools['goreturns'] = 'sourcegraph.com/sqs/goreturns';
+	}
+
+	// golint is no longer supported in go1.5
+	if (goVersion && (goVersion.major > 1 || (goVersion.major === 1 && goVersion.minor > 5))) {
+		tools['golint'] = 'github.com/golang/lint/golint';
+	}
+	return tools;
+}
 
 export function installAllTools() {
-	installTools(Object.keys(tools));
+	getGoVersion().then(() => installTools(Object.keys(getTools())));
```

Comment:
The version is used internally by `getTools()` and not `installTools` directly.

Refactored `installedTools` to not repeatedly call `getTools()` for each missing tool